dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```
@@ -0,0 +1,795 @@
+
+    /// <summary>Collection of properties to operate in SQLite data table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _dataTableProperties;
+
+    /// <summary>Collection of properties to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _vectorTableProperties;
+
+    /// <summary>Collection of property names to operate in SQLite data table.</summary>
+    private readonly Lazy<List<string>> _dataTableStoragePropertyNames;
+
+    /// <summary>Collection of property names to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<string>> _vectorTableStoragePropertyNames;
+
+    /// <summary>Table name in SQLite for data properties.</summary>
+    private readonly string _dataTableName;
+
+    /// <summary>Table name in SQLite for vector properties.</summary>
+    private readonly string _vectorTableName;
+
+    /// <inheritdoc />
+    public string CollectionName { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> class.
+    /// </summary>
+    /// <param name="connection"><see cref="DbConnection"/> that will be used to manage the data in SQLite.</param>
+    /// <param name="collectionName">The name of the collection/table that this <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> will access.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public SqliteVectorStoreRecordCollection(
+        DbConnection connection,
+        string collectionName,
+        SqliteVectorStoreRecordCollectionOptions<TRecord>? options = default)
+    {
+        // Verify.
+        Verify.NotNull(connection);
+        Verify.NotNullOrWhiteSpace(collectionName);
+        VectorStoreRecordPropertyVerification.VerifyGenericDataModelKeyType(typeof(TRecord), customMapperSupplied: false, SqliteConstants.SupportedKeyTypes);
```

`customMapperSupplied` should be checking if a custom mapper was supplied, instead of being hardcoded, otherwise it'll throw for cases where one was supplied and the generic data model is used.

---

Good catch, thanks!